docs/user-journeys/users/application-developer/foundational.md

```diff
@@ -9,80 +9,151 @@ track: "USERS › APPLICATION DEVELOPER › FOUNDATIONAL"
+#### Configuration
 
-## How Kubernetes works: an introduction
+To provide configuration data for your application, you can specify Kubernetes environment variables as part of the container definition in your Deployment manifest. However, this may require you to rebuild your container or modify your Deployment.
 
-To work with Kubernetes, you describe your cluster's desired state in terms of Kubernetes API objects. Cluster state includes but is not limited to the following information:
+To decouple configuration data from your workload, you can also use the following objects:
+
+* **{% glossary_tooltip text="ConfigMaps" term_id="configmap" %}** - For non-confidential data, such as an nginx-specific configuration file.
+
+* **{% glossary_tooltip text="Secrets" term_id="secret" %}** - Similar to ConfigMaps, but for confidential data.
+
+*ConfigMaps and Secrets are not overlapping options.* If you have *any* data that you want to keep private, you should be using a Secret. Otherwise there is nothing stopping that data from being exposed to malicious users.
+
+## Understand Kubernetes basics
+
+As an app developer, you don't need to know everything about the inner workings of Kubernetes, but you may find it helpful to understand it at a high-level.
+
+#### Overview
+
+Say that your team is deploying an ordinary Rails application. You've run some calculations and determined that you need five instances of your app running at any given time, in order to handle external traffic.
+
+If you're not running Kubernetes or a similar automated system, you might find the following scenario familiar:
+1. One instance of your app (a complete machine instance or just a container) goes down.
+2. Because your team has monitoring set up, this pages the person on-call.
+3. The on-call has to go in, investigate, and manually spin up a new instance.
+4. Depending how your team handles DNS/networking, the on-call may also need to also update the service discovery mechanism to point at the IP of the new Rails instance rather than the old.
+
+This process can be tedious and also inconvenient, especially if (2) happens in the early hours of the morning!
+
+**If you have Kubernetes set up, however, manual intervention is not as necessary.** Kubernetes will gracefully handle (3) and (4) on your behalf. As a a result, it's often referred to as a *self-healing* system.
+
+On a high-level, there are two key concepts to making this work: the *Kubernetes API* and the *Control Plane*.
+
+#### Kubernetes API
+
+For Kubernetes to be useful, it needs to know *what* sort of cluster state you want it to maintain. The {% glossary_tooltip text="Kubernetes API" term_id="kubernetes-api" %} provides a way for you to declare this desired state, in terms of API objects (such as {% glossary_tooltip text="Deployments" term_id="deployment" %}). Cluster state includes but is not limited to the following information:
```

Comment:
I don't think Kubernetes API and Control plane are the correct features to list for self healing. I see it more this way:

* Configuration files
* Control loops

I'll revisit this comment with a more concrete suggestion.